tests/error/comprehension_errors/guarded2.err

```diff
@@ -1,7 +1,11 @@
-Guppy compilation failed. Error in file $FILE:18
+Error: Linearity violation (at $FILE:18:22)
+   | 
+16 | @guppy(module)
+17 | def foo(qs: list[tuple[bool, qubit]] @owned) -> list[int]:
+18 |     return [42 for b, q in qs if b if bar(q)]
```

Comment:
does python allow chained `if`s? expected an `and`

Reply:
Yep, that works in Python

We don't support `and` and `or` in comprehensions yet since we compile them to dataflow...